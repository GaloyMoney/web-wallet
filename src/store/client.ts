import { ApolloClient, InMemoryCache, ApolloLink, from, HttpLink } from "@apollo/client"

import config from "server/config"
import { getCachedAuthToken, setCachedAuthToken } from "./use-auth-token"

export const cache = new InMemoryCache().restore(window.__G_DATA.ssrData)

const authLink = new ApolloLink((operation, forward) => {
  const user = getCachedAuthToken(cache)
  operation.setContext(({ headers }: { headers: Record<string, string> }) => ({
    headers: {
      authorization: user ? `Bearer ${user.authToken}` : "",
      ...headers,
    },
  }))
  return forward(operation)
})

const httpLink = new HttpLink({ uri: config.graphqlUri })

const client = new ApolloClient({
  cache,
  link: from([authLink, httpLink]),
})

if (window.__G_DATA.initialState.authToken) {
  setCachedAuthToken(client)(window.__G_DATA.initialState.authToken)
}

export default client
