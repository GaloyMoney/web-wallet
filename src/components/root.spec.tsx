import "@testing-library/jest-dom/extend-expect"

import { render } from "@testing-library/react"
import { MockedProvider } from "@galoymoney/client"

import RootComponent from "./root-component"

jest.mock("../kratos", () => ({
  KratosSdk: () => ({
    initializeSelfServiceRegistrationFlowForBrowsers: () =>
      Promise.resolve({ data: undefined }),
    initializeSelfServiceLoginFlowForBrowsers: () => Promise.resolve({ data: undefined }),
  }),
  handleFlowError: () => ({}),
}))

describe("Root appRoutes", () => {
  it("renders Home and matches snapshot", () => {
    const { asFragment } = render(
      <MockedProvider>
        <RootComponent path="/" />
      </MockedProvider>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders Send and matches snapshot", () => {
    const { asFragment } = render(
      <MockedProvider>
        <RootComponent path="/send" />
      </MockedProvider>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders Send with a destination and matches snapshot", () => {
    const { asFragment } = render(
      <MockedProvider>
        <RootComponent path="/send" to="user0" />
      </MockedProvider>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders Receive and matches snapshot", () => {
    const { asFragment } = render(
      <MockedProvider>
        <RootComponent path="/receive" />
      </MockedProvider>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders Contacts and matches snapshot", () => {
    const { asFragment } = render(
      <MockedProvider>
        <RootComponent path="/contacts" />
      </MockedProvider>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders Transactions and matches snapshot", () => {
    const { asFragment } = render(
      <MockedProvider>
        <RootComponent path="/transactions" username="user0" />
      </MockedProvider>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders Settings and matches snapshot", () => {
    const { asFragment } = render(
      <MockedProvider>
        <RootComponent path="/settings" />
      </MockedProvider>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})

describe("Root authRoutes", () => {
  it("renders Login and matches snapshot", () => {
    const { asFragment } = render(
      <MockedProvider>
        <RootComponent path="/login" />
      </MockedProvider>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it("renders Register and matches snapshot", () => {
    const { asFragment } = render(
      <MockedProvider>
        <RootComponent path="/register" />
      </MockedProvider>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
