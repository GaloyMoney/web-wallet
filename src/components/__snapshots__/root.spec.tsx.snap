// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Root appRoutes renders Contacts and matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-language="DEFAULT"
    id="main-container"
  >
    <div
      class="contacts"
    >
      <div
        class="header-container contacts-header"
      >
        <div
          class="header"
        >
          <div
            class="balance"
          >
            <div
              class="title"
            >
              Current Balance
            </div>
            <div
              class="value"
            >
              <div
                class="primary"
              >
                <i
                  aria-hidden="true"
                  class="fak fa-satoshisymbol-solid sat-symbol"
                />
                0
              </div>
              <div
                class="secondary"
              >
                $0.00
              </div>
            </div>
          </div>
          <div
            class="links"
          >
            <a
              href="/"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-home"
              />
              <span
                class="name"
              >
                Home
              </span>
            </a>
            <div
              class="separator"
            >
              |
            </div>
            <a
              href="/login"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-sign-in-alt"
              />
              <span
                class="name"
              >
                Login
              </span>
            </a>
            <div
              class="separator"
            >
              |
            </div>
            <div
              class="menu-icon"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-bars"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="page-title"
      >
        Contacts
      </div>
      <div
        class="no-data"
      >
        No Contacts
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Root appRoutes renders Home and matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-language="DEFAULT"
    id="main-container"
  >
    <div
      class="home"
    >
      <div
        class="header-container home-header"
      >
        <div
          class="header"
        >
          <div
            class="balance"
          >
            <div
              class="title"
            >
              Current Balance
            </div>
            <div
              class="value"
            >
              <div
                class="primary"
              >
                <i
                  aria-hidden="true"
                  class="fak fa-satoshisymbol-solid sat-symbol"
                />
                0
              </div>
              <div
                class="secondary"
              >
                $0.00
              </div>
            </div>
          </div>
          <div
            class="links"
          >
            <a
              href="/login"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-sign-in-alt"
              />
              <span
                class="name"
              >
                Login
              </span>
            </a>
            <div
              class="separator"
            >
              |
            </div>
            <div
              class="menu-icon"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-bars"
              />
            </div>
          </div>
        </div>
        <div
          class="header-nav"
        >
          <a
            href="/scan"
          >
            <i
              aria-hidden="true"
              class="fa-solid fa-qrcode"
            />
            Scan QR code
          </a>
          <a
            class="link"
            href="/send"
          >
            <svg
              fill="none"
              height="58"
              viewBox="0 0 58 58"
              width="58"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M55.569 29c1.342 0 2.442 1.09 2.33 2.429A29 29 0 1 1 28.898 0c1.342-.005 2.34 1.179 2.233 2.517-.108 1.339-1.282 2.323-2.625 2.35a24.137 24.137 0 1 0 24.508 26.56C53.15 30.091 54.225 29 55.569 29z"
                fill="#ed4b85"
              />
              <g
                clip-path="url(#e21ubekeva)"
                fill="#ed4b85"
                stroke="#ed4b85"
                stroke-width=".5"
              >
                <path
                  d="M30.163 14.236h-2.325v3.97h2.325v-3.97zM30.163 39.862h-2.325v3.971h2.325v-3.97zM37.287 24.017v-2.324H20.715v2.324h16.572zM37.287 30.156v-2.324H20.715v2.324h16.572zM37.287 36.12v-2.323H20.715v2.324h16.572z"
                />
              </g>
              <g
                clip-path="url(#3gzwouqxvb)"
              >
                <path
                  d="m42.972 1.628-.038.004V2.68c0 .72.66 1.379 1.38 1.379h7.054L40.066 15.326a1.277 1.277 0 0 0 0 1.821l.783.779c.51.51 1.293.507 1.803-.002L53.95 6.578v7.139c0 .72.676 1.358 1.397 1.358h1.107c.72 0 1.23-.638 1.23-1.358V1.577c0-.349-.099-.648-.347-.895a1.183 1.183 0 0 0-.887-.357H44.314c-.72 0-1.342.583-1.342 1.303z"
                  fill="#ed4b85"
                />
              </g>
              <defs>
                <clippath
                  id="e21ubekeva"
                >
                  <path
                    d="M0 0h30v30H0z"
                    fill="#fff"
                    transform="translate(14 14)"
                  />
                </clippath>
                <clippath
                  id="3gzwouqxvb"
                >
                  <path
                    d="M0 0h18v18H0z"
                    fill="#fff"
                    transform="rotate(-90 29 -10.684)"
                  />
                </clippath>
              </defs>
            </svg>
            Send Bitcoin
          </a>
          <a
            class="link"
            href="/receive"
          >
            <svg
              fill="none"
              height="58"
              viewBox="0 0 58 58"
              width="58"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M55.569 29c1.342 0 2.442 1.09 2.33 2.429A29 29 0 1 1 28.898 0c1.342-.005 2.34 1.179 2.233 2.517-.108 1.339-1.282 2.323-2.625 2.35a24.137 24.137 0 1 0 24.508 26.56C53.15 30.091 54.225 29 55.569 29z"
                fill="#28bf89"
              />
              <g
                clip-path="url(#m0urlq4ysa)"
                fill="#28bf89"
                stroke="#28bf89"
                stroke-width=".5"
              >
                <path
                  d="M30.408 14.481h-2.324v3.971h2.324v-3.97zM30.408 40.108h-2.324v3.971h2.324v-3.97zM37.532 24.263V21.94H20.961v2.324h16.571zM37.532 30.401v-2.324H20.961v2.324h16.571zM37.532 36.367v-2.325H20.961v2.325h16.571z"
                />
              </g>
              <g
                clip-path="url(#ebebktty6b)"
              >
                <path
                  d="m54.712 16.688.038-.004v-1.048c0-.721-.66-1.379-1.38-1.379h-7.054L57.619 2.99a1.277 1.277 0 0 0 0-1.821L56.836.39a1.255 1.255 0 0 0-1.804.002L43.734 11.737V4.6c0-.72-.676-1.358-1.397-1.358H41.23c-.72 0-1.23.638-1.23 1.358v12.14c0 .348.099.648.347.895.248.25.538.357.887.357H53.37c.72 0 1.342-.583 1.342-1.303z"
                  fill="#28bf89"
                />
              </g>
              <defs>
                <clippath
                  id="m0urlq4ysa"
                >
                  <path
                    d="M0 0h30v30H0z"
                    fill="#fff"
                    transform="translate(14.246 14.246)"
                  />
                </clippath>
                <clippath
                  id="ebebktty6b"
                >
                  <path
                    d="M0 0h18v18H0z"
                    fill="#fff"
                    transform="rotate(90 29 29)"
                  />
                </clippath>
              </defs>
            </svg>
            Receive Bitcoin
          </a>
        </div>
      </div>
      <div
        class="recent-transactions"
      />
    </div>
    <div
      id="powered-by"
    >
      <div
        class="content"
      >
        Powered By 
        <a
          href="https://galoy.io/"
          rel="noreferrer"
          target="_blank"
        >
          Galoy
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Root appRoutes renders Receive and matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-language="DEFAULT"
    id="main-container"
  >
    <div
      class="receive"
    >
      <div
        class="header-container receive-bitcoin-header"
      >
        <div
          class="header"
        >
          <div
            class="balance"
          >
            <div
              class="title"
            >
              Current Balance
            </div>
            <div
              class="value"
            >
              <div
                class="primary"
              >
                <i
                  aria-hidden="true"
                  class="fak fa-satoshisymbol-solid sat-symbol"
                />
                0
              </div>
              <div
                class="secondary"
              >
                $0.00
              </div>
            </div>
          </div>
          <div
            class="links"
          >
            <a
              href="/"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-home"
              />
              <span
                class="name"
              >
                Home
              </span>
            </a>
            <div
              class="separator"
            >
              |
            </div>
            <a
              href="/login"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-sign-in-alt"
              />
              <span
                class="name"
              >
                Login
              </span>
            </a>
            <div
              class="separator"
            >
              |
            </div>
            <div
              class="menu-icon"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-bars"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="page-title"
      >
        Receive Bitcoin
      </div>
      <div
        class="tabs"
      >
        <div
          class="tab active"
        >
          Lightning
        </div>
        <div
          class="tab link"
        >
          OnChain
        </div>
      </div>
      <div
        class="tab-content"
      >
        <div
          class="amount-input center-display"
        >
          <div
            class="input-label"
          >
            Set Amount
          </div>
          <div
            class="amount-input-form"
          >
            <div
              class="currency-label"
            >
              $
            </div>
            <input
              inputmode="numeric"
              pattern="\\\\d*"
              placeholder="Set invoice value in USD"
              type="text"
              value=""
            />
            <div
              class="toggle-currency link"
            >
              ⇅
            </div>
          </div>
        </div>
        <div
          class="note-input center-display"
        >
          <div
            class="input-label"
          >
            Note or Label
          </div>
          <textarea
            name="memo"
            placeholder="Set a note for the sender here (optional)"
            rows="3"
          />
        </div>
        <div
          class="action-button center-display"
        >
          <button
            disabled=""
          >
            Next 
          </button>
        </div>
        <div
          class="action-button center-display"
        >
          <a
            href="/"
          >
            Cancel
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Root appRoutes renders Send and matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-language="DEFAULT"
    id="main-container"
  >
    <div
      class="send"
    >
      <div
        class="header-container send-bitcoin-header"
      >
        <div
          class="header"
        >
          <div
            class="balance"
          >
            <div
              class="title"
            >
              Current Balance
            </div>
            <div
              class="value"
            >
              <div
                class="primary"
              >
                <i
                  aria-hidden="true"
                  class="fak fa-satoshisymbol-solid sat-symbol"
                />
                0
              </div>
              <div
                class="secondary"
              >
                $0.00
              </div>
            </div>
          </div>
          <div
            class="links"
          >
            <a
              href="/"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-home"
              />
              <span
                class="name"
              >
                Home
              </span>
            </a>
            <div
              class="separator"
            >
              |
            </div>
            <a
              href="/login"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-sign-in-alt"
              />
              <span
                class="name"
              >
                Login
              </span>
            </a>
            <div
              class="separator"
            >
              |
            </div>
            <div
              class="menu-icon"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-bars"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="page-title"
      >
        Send Bitcoin
      </div>
      <div
        class="destination-input center-display"
      >
        <div
          class="input-label"
        >
          To Recipient
        </div>
        <div
          class="destination"
        >
          <input
            autocomplete="off"
            name="destination"
            placeholder="username, invoice, or address"
            type="text"
            value=""
          />
          <div
            class="qrcode-detector"
          >
            <div
              class="scan-start link"
            >
              <i
                aria-hidden="true"
                class="fas fa-qrcode"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="action-button center-display"
      >
        <button
          disabled=""
        >
          Next 
        </button>
      </div>
      <div
        class="action-button center-display"
      >
        <a
          href="/"
        >
          Cancel
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Root appRoutes renders Send with a destination and matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-language="DEFAULT"
    id="main-container"
  >
    <div
      class="send"
    >
      <div
        class="header-container send-bitcoin-header"
      >
        <div
          class="header"
        >
          <div
            class="balance"
          >
            <div
              class="title"
            >
              Current Balance
            </div>
            <div
              class="value"
            >
              <div
                class="primary"
              >
                <i
                  aria-hidden="true"
                  class="fak fa-satoshisymbol-solid sat-symbol"
                />
                0
              </div>
              <div
                class="secondary"
              >
                $0.00
              </div>
            </div>
          </div>
          <div
            class="links"
          >
            <a
              href="/"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-home"
              />
              <span
                class="name"
              >
                Home
              </span>
            </a>
            <div
              class="separator"
            >
              |
            </div>
            <a
              href="/login"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-sign-in-alt"
              />
              <span
                class="name"
              >
                Login
              </span>
            </a>
            <div
              class="separator"
            >
              |
            </div>
            <div
              class="menu-icon"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-bars"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="page-title"
      >
        Send Bitcoin
      </div>
      <div
        class="destination-input center-display"
      >
        <div
          class="input-label"
        >
          To Recipient
        </div>
        <div
          class="destination"
        >
          <input
            autocomplete="off"
            name="destination"
            placeholder="username, invoice, or address"
            type="text"
            value="user0"
          />
          <div
            class="qrcode-detector"
          >
            <div
              class="scan-start link"
            >
              <i
                aria-hidden="true"
                class="fas fa-qrcode"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="action-button center-display"
      >
        <button>
          Next 
        </button>
      </div>
      <div
        class="action-button center-display"
      >
        <a
          href="/"
        >
          Cancel
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Root appRoutes renders Settings and matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-language="DEFAULT"
    id="main-container"
  >
    <div
      class="settings"
    >
      <div
        class="header-container settings-header"
      >
        <div
          class="header"
        >
          <div
            class="balance"
          >
            <div
              class="title"
            >
              Current Balance
            </div>
            <div
              class="value"
            >
              <div
                class="primary"
              >
                <i
                  aria-hidden="true"
                  class="fak fa-satoshisymbol-solid sat-symbol"
                />
                0
              </div>
              <div
                class="secondary"
              >
                $0.00
              </div>
            </div>
          </div>
          <div
            class="links"
          >
            <a
              href="/"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-home"
              />
              <span
                class="name"
              >
                Home
              </span>
            </a>
            <div
              class="separator"
            >
              |
            </div>
            <a
              href="/login"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-sign-in-alt"
              />
              <span
                class="name"
              >
                Login
              </span>
            </a>
            <div
              class="separator"
            >
              |
            </div>
            <div
              class="menu-icon"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-bars"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="list"
      >
        <div
          class="setting"
        >
          <div
            class="icon"
          >
            <i
              aria-hidden="true"
              class="fa-solid fa-at"
            />
          </div>
          <div
            class="name"
          >
            Email
            <div
              class="sub"
            >
              (not logged in)
            </div>
          </div>
          <div
            class="action"
          >
            <i
              aria-hidden="true"
              class="fa-solid fa-lock"
            />
          </div>
        </div>
        <div
          class="setting"
        >
          <div
            class="icon"
          >
            <i
              aria-hidden="true"
              class="fa-solid fa-language"
            />
          </div>
          <div
            class="name"
          >
            Language
            <div
              class="sub"
            >
              Default (OS)
            </div>
          </div>
          <div
            class="action"
          >
            <select
              disabled=""
              name="language"
            >
              <option
                value="DEFAULT"
              >
                Default (OS)
              </option>
              <option
                value="en-US"
              >
                English
              </option>
              <option
                value="es-SV"
              >
                Spanish
              </option>
            </select>
          </div>
        </div>
        <div
          class="setting"
        >
          <div
            class="icon"
          >
            <i
              aria-hidden="true"
              class="fa-solid fa-palette"
            />
          </div>
          <div
            class="name"
          >
            Color Theme
            <div
              class="sub"
            >
              Default (OS)
            </div>
          </div>
          <div
            class="action"
          >
            <select
              name="colorTheme"
            >
              <option
                value="DEFAULT"
              >
                Default (OS)
              </option>
              <option
                value="dark"
              >
                Dark mode
              </option>
              <option
                value="light"
              >
                Light mode
              </option>
            </select>
          </div>
        </div>
        <div
          class="setting"
        >
          <a
            href="/login"
          >
            <i
              aria-hidden="true"
              class="fa-solid fa-sign-in-alt"
            />
            <span
              class="name"
            >
              Login
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Root appRoutes renders Transactions and matches snapshot 1`] = `
<DocumentFragment>
  <div
    data-language="DEFAULT"
    id="main-container"
  >
    <div
      class="transaction-list"
    >
      <div
        class="header-container transactions-header"
      >
        <div
          class="header"
        >
          <div
            class="balance"
          >
            <div
              class="title"
            >
              Current Balance
            </div>
            <div
              class="value"
            >
              <div
                class="primary"
              >
                <i
                  aria-hidden="true"
                  class="fak fa-satoshisymbol-solid sat-symbol"
                />
                0
              </div>
              <div
                class="secondary"
              >
                $0.00
              </div>
            </div>
          </div>
          <div
            class="links"
          >
            <a
              href="/"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-home"
              />
              <span
                class="name"
              >
                Home
              </span>
            </a>
            <div
              class="separator"
            >
              |
            </div>
            <a
              href="/login"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-sign-in-alt"
              />
              <span
                class="name"
              >
                Login
              </span>
            </a>
            <div
              class="separator"
            >
              |
            </div>
            <div
              class="menu-icon"
            >
              <i
                aria-hidden="true"
                class="fa-solid fa-bars"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="page-title"
      >
        Transactions with user0
      </div>
      <div
        class="load-more"
      >
        <i
          aria-hidden="true"
          class="spinner fas fa-spinner fa-spin fa-lg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Root authRoutes renders Login and matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="main-container"
  >
    <div
      class="login-form auth-form"
    >
      <form
        method="POST"
      >
        <input
          name="csrf_token"
          type="hidden"
          value=""
        />
        <div
          class="input-container"
        >
          <div
            class=""
          >
            Email
          </div>
          <input
            autocomplete="email"
            name="identifier"
            required=""
            type="email"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <div
            class=""
          >
            Password
          </div>
          <input
            autocomplete="current-password"
            name="password"
            required=""
            type="password"
          />
        </div>
        <div
          class="button-container"
        >
          <button
            class="button"
            name="method"
            value="password"
          >
            Login
          </button>
          <a
            href="/"
          >
            Cancel
          </a>
        </div>
      </form>
    </div>
    <div
      class="form-links"
    >
      <a
        href="/register"
      >
        <i
          aria-hidden="true"
          class="fas fa-sign-in-alt"
        />
        Create new account
      </a>
      <div
        class="separator"
      >
        |
      </div>
      <a
        href="/recovery"
      >
        <i
          aria-hidden="true"
          class="fas fa-key"
        />
        Recover your account
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Root authRoutes renders Recorver and matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="main-container"
  >
    <div
      class="recovery-form auth-form"
    >
      <form
        method="POST"
      >
        <input
          name="csrf_token"
          type="hidden"
          value=""
        />
        <div
          class="input-container"
        >
          <div
            class=""
          >
            Email
          </div>
          <input
            autocomplete="email"
            name="email"
            required=""
            type="email"
          />
        </div>
        <div
          class="button-container"
        >
          <button
            class="button"
            name="method"
            value="link"
          >
            Recover Account
          </button>
          <a
            href="/"
          >
            Cancel
          </a>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Root authRoutes renders Register and matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="main-container"
  >
    <div
      class="register-form auth-form"
    >
      <form
        method="POST"
      >
        <input
          name="csrf_token"
          type="hidden"
          value=""
        />
        <div
          class="input-container"
        >
          <div
            class=""
          >
            Email
          </div>
          <input
            autocomplete="email"
            name="traits.email"
            required=""
            type="email"
            value=""
          />
        </div>
        <div
          class="input-container"
        >
          <div
            class=""
          >
            Password
          </div>
          <input
            autocomplete="current-password"
            name="password"
            required=""
            type="password"
          />
        </div>
        <div
          class="button-container"
        >
          <button
            class="button"
            name="method"
            value="password"
          >
            Create Account
          </button>
          <a
            href="/"
          >
            Cancel
          </a>
        </div>
      </form>
    </div>
    <div
      class="form-links"
    >
      <a
        href="/login"
      >
        <i
          aria-hidden="true"
          class="fas fa-sign-in-alt"
        />
        Login
      </a>
    </div>
  </div>
</DocumentFragment>
`;
